src/terms/anyonic.jl

```diff
@@ -2,14 +2,60 @@
 # This term does not contain the kinetic energy, which must be added separately
 # /!\ They have no 1/2 factor in front of the kinetic energy,
 #     so for consistency the added kinetic energy must have a scaling of 2
-# Energy = <u, ((-i∇ + βA)^2 + V) u>
+# Energy = <u, ((-i hbar ∇ + βA)^2 + V) u>
 # where ∇∧A = 2π ρ, ∇⋅A = 0 => A = x^⟂/|x|² ∗ ρ
-# H = (-i∇ + βA)² + V - 2β x^⟂/|x|² ∗ (βAρ + J)
-#   = -Δ + 2β (-i∇)⋅A + β²|A|^2 - 2β x^⟂/|x|² ∗ (βAρ + J)
+# H = (-i hbar ∇ + βA)² + V - 2β x^⟂/|x|² ∗ (βAρ + J)
+#   = - hbar^2 Δ + 2β hbar (-i∇)⋅A + β²|A|^2 - 2β x^⟂/|x|² ∗ (βAρ + J)
 # where only the first three terms "count for the energy", and where
-# J = 1/(2i) (u* ∇u - u ∇u*)
+# J = 1/(2i) hbar (u* ∇u - u ∇u*)
+
+# for numerical reasons, we solve the equation ∇∧A = 2π ρ in two parts: A = A_SR + A_LR, with
+# ∇∧A_SR = 2π(ρ-ρref)
+# ∇∧A_LR = 2πρref
+# where ρref is a gaussian centered at the origin and with the same integral as ρ (=M)
+# ρref(x) = M exp(-1/2 (x/σ)^2) / (σ^3 (2π)^3/2)
+# This way, ρ-ρref has zero mass, and A_SR is shorter range. The long-range part is computed explicitly
+
+function ρref_real(x::T, y::T, M, σ) where {T <: Real}
+    r = hypot(x, y)
+    M * exp(-1/2 * (r/σ)^2) / (σ^2 * (2T(π))^(2/2))
+end
+
+function magnetic_field_produced_by_ρref(x::T, y::T, M, σ) where {T <: Real}
+    # The solution of ∇∧A = 2π ρref is ϕ(r) [-y;x] where ϕ satisfies
+    # ∇∧A = 2ϕ + r ϕ' => rϕ' + 2 ϕ = 2π ρref
+    # Wolfram alpha (after a bit of coaxing) says the solution to
+    # rϕ' + 2ϕ = C exp(-α x^2)
+    # is
+    # 1/x^2 (c1 - C exp(-α x^2)/2α), which coupled with smoothness at 0 gives
+    # C/(2α)/x^2*(1 - exp(-α x^2))
+    r = hypot(x, y)
+    r == 0 && return magnetic_field_produced_by_ρref(1e-8, 0.0, M, σ) # hack
+
+    α = 1/(2*σ^2)
+    C = 2T(π)*M/(σ^2 * (2T(π))^(2/2))
+    r = hypot(x, y)
+    ϕ = 1/2*C/α/r^2*(1-exp(-α*r^2))
+    ϕ * @SVector[-y, x]
+end
+
+function make_div_free(basis::PlaneWaveBasis{T}, A) where {T}
```

Comment:
A comment for what this function does: Perhaps just the formula written out more nicely.

Reply:
The name is self-explanatory, no? I could write a comment, but the comment is going to be "make a given vector field divergence-free", which does not actually add more information. I did add a comment on why this was necessary

Reply:
Just write the formula in nice physics instead of G's and ffts.